package/MDAnalysis/coordinates/timestep.pyx

```diff
@@ -0,0 +1,982 @@
+            raise ValueError("Must specify at least one set of data")
+        n_atoms = max(lens)
+        # Check arrays are matched length?
+        if not all(val == n_atoms for val in lens):
+            raise ValueError("Lengths of input data mismatched")
+
+        ts = cls(n_atoms,
+                 positions=has_positions,
+                 velocities=has_velocities,
+                 forces=has_forces,
+                 **kwargs)
+        if has_positions:
+            ts.positions = positions
+        if has_velocities:
+            ts.velocities = velocities
+        if has_forces:
+            ts.forces = forces
+
+        return ts
+
+    def __eq__(self, other):
+        """Compare with another Timestep
+
+        .. versionadded:: 0.11.0
+        """
+        if not isinstance(other, Timestep):
+            return False
+
+        if not self.frame == other.frame:
+            return False
+
+        if not self.n_atoms == other.n_atoms:
+            return False
+
+        if not self.has_positions == other.has_positions:
+            return False
+        if self.has_positions:
+            if not (self.positions == other.positions).all():
+                return False
+
+        if self.dimensions is None:
+            if other.dimensions is not None:
+                return False
+        else:
+            if other.dimensions is None:
+                return False
+            if not (self.dimensions == other.dimensions).all():
+                return False
+
+        if not self.has_velocities == other.has_velocities:
+            return False
+        if self.has_velocities:
+            if not (self.velocities == other.velocities).all():
+                return False
+
+        if not self.has_forces == other.has_forces:
+            return False
+        if self.has_forces:
+            if not (self.forces == other.forces).all():
+                return False
+
+        return True
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __getitem__(self, atoms):
+        """Get a selection of coordinates
+
+        ``ts[i]``
+
+           return coordinates for the i'th atom (0-based)
+
+        ``ts[start:stop:skip]``
+
+           return an array of coordinates, where start, stop and skip
+           correspond to atom indices,
+           :attr:`MDAnalysis.core.groups.Atom.index` (0-based)
+        """
+        if isinstance(atoms, numbers.Integral):
+            return self._pos[atoms]
+        elif isinstance(atoms, (slice, np.ndarray)):
+            return self._pos[atoms]
+        else:
+            raise TypeError
+
+    def __getattr__(self, attr):
+        # special-case timestep info
+        if attr in ('velocities', 'forces', 'positions'):
+            raise NoDataError('This Timestep has no ' + attr)
+        err = "{selfcls} object has no attribute '{attr}'"
+        raise AttributeError(err.format(selfcls=type(self).__name__,
+                                        attr=attr))
```

Comment:
I can't remember why we have this... Is it still necessary now that `.positions` etc are properties?

Reply:
Depends if you want a custom error message for attribute error, but I agree that it doesn't serve its original purpose as `velocities, forces ...` are always present. 

Should I remove?

Reply:
I am going to remove this as it no longer serves a useful purpose with attributes being always present.